apps/emqx_rule_engine/src/emqx_rule_monitor.erl

```diff
@@ -45,18 +46,21 @@ init([]) ->
     _ = erlang:process_flag(trap_exit, true),
     {ok, #{retryers => #{}}}.
 
+async_refresh_resources_rules() ->
+    gen_server:cast(?MODULE, {create_handler, refresh, resources_and_rules, #{}}).
```

Comment:
I don't understand why we have to re-use the retry_loop to do this.

```suggestion
async_refresh_resources_rules() ->
    gen_server:cast(?MODULE, async_refresh).
```
and then
```
handle_cast(async_refresh, State) ->
    Pid = spawn_link(fun do_async_refresh/0),
    {no_reply, State#{boot_refresh_pid => Pid}};
```

and then

```
do_async_refresh() ->
    %% NOTE: the order matters.
    %% We should always refresh the resources first and then the rules.
    ok = emqx_rule_engine:refresh_resources(),
    ok = emqx_rule_engine:refresh_rules().
```

then handle the EXIT signal in another handle_info clause.

Reply:
Because this way we have to handle the EXIT signal again.

And the most import reason, is that the emqx_rule_monitor module was designed to manage different retriers or async tasks. So when creating a `handler`, it requires the caller to provide the `Tag` as the task type and `Obj` as the task Id.

I just to improve the design when fixing this issue.

Reply:
I've changed the code as recommended.